src/main/java/io/xpring/xrpl/model/XRPCurrencyAmount.java

```diff
@@ -0,0 +1,62 @@
+package io.xpring.xrpl.model;
+
+import org.xrpl.rpc.v1.CurrencyAmount;
+import org.immutables.value.Value;
+import org.xrpl.rpc.v1.IssuedCurrencyAmount;
+
+import javax.annotation.Nullable;
+
+/**
+ * An amount of currency on the XRP Ledger
+ * @see "https://xrpl.org/basic-data-types.html#specifying-currency-amounts"
+ */
+@Value.Immutable
+public interface XRPCurrencyAmount {
+    static ImmutableXRPCurrencyAmount.Builder builder() {
+        return ImmutableXRPCurrencyAmount.builder();
+    }
+
+    /**
+     * @return An amount of XRP, specified in drops.
+     * Mutually exclusive fields - only drops XOR issuedCurrency should be set.
+     */
+     @Nullable String drops();
+
+    /**
+     * @return An amount of an issued currency.
+     * Mutually exclusive fields - only drops XOR issuedCurrency should be set.
+     */
+    @Nullable XRPIssuedCurrency issuedCurrency();
+
+    static XRPCurrencyAmount from(CurrencyAmount currencyAmount) {
+        switch (currencyAmount.getAmountCase()) {
+            // Mutually exclusive: either drops or issuedCurrency is set in an XRPCurrencyAmount
+            case ISSUED_CURRENCY_AMOUNT: {
+                IssuedCurrencyAmount issuedCurrencyAmount = currencyAmount.getIssuedCurrencyAmount();
+                XRPIssuedCurrency xrpIssuedCurrency;
+                if (issuedCurrencyAmount != null) {
+                    try {
+                        xrpIssuedCurrency = XRPIssuedCurrency.from(issuedCurrencyAmount);
+                    } catch (NumberFormatException error) {
```

Comment:
You shouldn't need to rethrow the error here, I don't think.  If `XRPIssuedCurrency.from` throws an exception, this method will throw the exception.

Also, I know @keefertaylor hates 1 letter names, but it's more normal in Java to do `NumberFormatException e`.  If that really bugs you, let's at least change `error` to `exception` (Not here if this is going away, but more for future reference)